src/core/Spy.js

```diff
@@ -0,0 +1,58 @@
+getJasmineRequireObj().Spy = function (j$) {
+
+  function Spy(name, originalFn) {
+    var args = buildArgs(),
+      /*`eval` is the only option to preserve both this and context:
+        - former is needed to work as expected with methods,
+        - latter is needed to access real spy function and allows to reduce eval'ed code to absolute minimum
+        More explanation here (look at comments): http://www.bennadel.com/blog/1909-javascript-function-constructor-does-not-create-a-closure.htm
+       */
+      /* jshint evil: true */
+      wrapper = eval('(function (' + args + ') { return spy.apply(this, Array.prototype.slice.call(arguments)); })'),
```

Comment:
Dang. After playing around with it for a while, I now better understand exactly what you mean — `Function` does not create closures. You can sort of force the `spy` inside the function body if you `bind(spy)` and use `this.apply` but then you can't change the binding of the returned spy. Basically it's just not possible to do without `eval`, as you already figured out. Or you can hard-code in the proxies like Sinon does but then you should add documentation to the effect of "Jasmine spies preserve arity up to length XX but not longer" or similar.

How frustrating.